src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import functions.*;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Duke {
+    public static void main(String[] args) {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        System.out.println("Hello Hello! I'm Duke\n" + "What can I do for you?");
+        int i = 0;
+        String txt;
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            i = DukeSave.load(tasks);
+        } catch (Exception m) {
+            System.out.println("An exception has occurred: " + m);
+        }
+
+        while (true) {
+            txt = in.nextLine();
+            try {
+                if (txt.equals("list")) {                        //CMD LIST
+                    System.out.println("Here are the tasks in your list:");
+                    for (int x = 0; x < i; x++) {
+                        System.out.printf("%d.%s\n", x + 1, tasks.get(x));
+                    }
+                } else if (txt.startsWith("done ")) {            //CMD DONE
+                    if (txt.equals("done ")) {
+                        throw new DukeException("No description entered ...");
+                    }
+                    int num;
+                    num = Integer.parseInt(txt.substring(5));
+                    if (num > i || num < 1) {
+                        throw new DukeException("Invalid number entered ...");
+                    }
+
+                    tasks.get(num - 1).isDone = true;
+                    DukeSave.save(tasks, i);
+
+                    System.out.println("Nice! I've marked this task as done:");
+                    System.out.println(tasks.get(num - 1));
+                } else if (txt.startsWith("todo ")) {            //CMD TODO
+                    if (txt.equals("todo ")) {
+                        throw new DukeException("No description entered ...");
+                    }
+                    tasks.add(i++, new Todo(txt.substring(5)));
+                    DukeSave.save(tasks, i);
+
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println(tasks.get(i - 1));
+                    System.out.printf("Now you have %d tasks in the list.\n", i);
```

Comment:
This 3 lines seem to be repeated a lot. Perhaps you can abstract this out into a method and call it? This will make the code more readable and also easier to change.